src/connection/adapter/base_http_adapter.ts

```diff
@@ -83,14 +84,25 @@ function isDecompressionError(result: any): result is { error: Error } {
   return result.error !== undefined
 }
 
+const expiredSocketMessage = 'expired socket'
+
 export abstract class BaseHttpAdapter implements Connection {
   protected readonly headers: Http.OutgoingHttpHeaders
+  private readonly retry_strategy?: RetryStrategy
+  private readonly known_sockets = new Map<
+    string,
+    {
+      last_used_time: number
+      timeout_id?: NodeJS.Timeout
```

Comment:
This is a rudiment, as I was experimenting with background expiring idling sockets cleanup. Will remove.